sale_stock_refund_picking/README.rst

```diff
@@ -0,0 +1,95 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/lgpl-3.0-standalone.html
+   :alt: License: LGPL-3
+
+======================================
+Refund Return Pickings in Sales Orders
+======================================
+
+This module extends the functionality of sale orders to support marking some
+stock picking moves as "To refund in Sale Order" and to allow you to create
+invoices deducting the products that are returned and refunded.
+
+Usage
+=====
+
+To use this module, when some customer returns some refundable products to you
+before you created an invoice, you need to:
+
+#. Go to *Sales > Sales Orders > Create*.
+#. Choose a customer and add a product whose *Invoicing Policy* is *Delivered
+   quantities*, and input that you are selling 20 units.
+#. Confirm the sale.
+#. Go to *Delivery > Validate > Apply > Reverse*.
+#. Set *Quantity* to 5, and enable *To Refund*.
+#. Press *Return > Validate > Apply*.
+#. Return to the sale order.
+#. Press *Create Invoice > Invoiceable lines > Create and View Invoices*.
+#. The created invoice's amount is 15 units now.
+
+To use this module, when some customer returns some refundable products to you
+after you created an invoice, you need to:
+
+#. Go to *Sales > Sales Orders > Create*.
+#. Choose a customer and add a product whose *Invoicing Policy* is *Delivered
+   quantities*, and input that you are selling 20 units.
+#. Confirm the sale.
+#. Go to *Delivery > Validate > Apply*.
+#. Return to the sale order.
+#. Press *Create Invoice > Invoiceable lines > Create and View Invoices*.
+#. The created invoice's amount is 20 units now.
+#. Return to the sale order.
+#. Go to *Delivery > Reverse*.
+#. Set *Quantity* to 5, and enable *To Refund*.
+#. Press *Return > Validate > Apply*.
+#. Return to the sale order.
+#. Press *Create Invoice > Invoiceable lines (deduct down payments) >
+   Create and View Invoices*.
+#. The created invoice's amount is -5 units now.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/95/9.0
+
+Known issues / Roadmap
+======================
+
+* This addon is a pseudobackport of a functionality that exists natively in
+  v10, plus a fix for https://github.com/odoo/odoo/issues/13974, so this addon
```

Comment:
No, it applies, because it hides the field when the picking is not related to a purchase order; a different addon will implement this for POs.

I will tag the code as a HACK where it belongs, anyway.